easybuild/tools/modules.py

```diff
@@ -681,13 +681,42 @@ def modpath_extensions_for(self, mod_names):
         # copy environment so we can restore it
         env = os.environ.copy()
 
+        # regex for $MODULEPATH definitions in Python syntax, e.g.:
+        # os.environ['MODULEPATH'] = "/tmp/example:/home/example/modules:";
+        modpath_ext_regex = re.compile(r"^os\.environ\[.MODULEPATH.\]\s*=\s*(?:'|\")(\S+)(?:'|\");?", re.M)
+
         modpath_exts = {}
         for mod_name in mod_names:
-            modtxt = self.read_module_file(mod_name)
-            useregex = re.compile(r'^\s*module\s+use\s+"?([^"\s]+)"?', re.M)
-            exts = useregex.findall(modtxt)
 
-            self.log.debug("Found $MODULEPATH extensions for %s: %s" % (mod_name, exts))
+            # module must be available to (fake) load
+            if not self.exist([mod_name])[0]:
+                raise EasyBuildError("Can not determine $MODULEPATH extensions for non-existing module %s", mod_name)
+
+            # grab current value for $MODULEPATH, so we can figure out which paths get added to it
+            curr_modulepath = [p for p in os.environ.get('MODULEPATH', '').split(os.pathsep) if p]
+
+            # fake load for module: just obtain changes that would be made to environment, don't apply them (yet)
+            # this is prefrred over parsing the module file contents, to take into account conditional extensions
+            mod_load_txt = self.run_module('load', mod_name, return_output=True)
```

Comment:
PR updated, I reverted to actually parsing the module file itself, after figuring out that including `$MODULEPATH` extensions that may not actually happen at runtime due to a condition not being met is fine...